src/main/java/org/jboss/forge/arquillian/command/algeron/AbstractAlgeronRetrieverCommand.java

```diff
@@ -0,0 +1,81 @@
+package org.jboss.forge.arquillian.command.algeron;
+
+import org.jboss.forge.addon.dependencies.builder.DependencyBuilder;
+import org.jboss.forge.addon.facets.FacetFactory;
+import org.jboss.forge.addon.projects.ProjectFactory;
+import org.jboss.forge.addon.projects.ui.AbstractProjectCommand;
+import org.jboss.forge.addon.ui.command.UICommand;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.arquillian.api.algeron.AlgeronRetrieverFacet;
+import org.jboss.forge.arquillian.testframework.algeron.AlgeronProvider;
+
+import javax.inject.Inject;
+import java.util.Map;
+
+public abstract class AbstractAlgeronRetrieverCommand extends AbstractProjectCommand implements UICommand {
+    public static final DependencyBuilder NO_DEPENDENCY = null;
```

Comment:
Since I am using same approach in publisher, I am going to open an issue to refactor names and this pattern in both places, so for now both retriever and publisher uses same approach, and then we fix it in a reducing technical dept PR.